src/main/java/com/education/service/TimetableServiceImpl.java

```diff
@@ -92,48 +91,63 @@ public void update(long id, Timetable newTimetable) {
+     *    3-1-1 if Start timetable is before end of timetables and start of timetable is  after start of timetables or
      *          end of timetable is before end of timetables and end of timetable is after start of timetables
-     *      8-1-1-1 throws an exception
-     *    8-1-2 else if start of timetable is equal of start of timetables
-     *      8-1-2-1 throws an exception END IF
+     *      3-1-1-1 throws an exception
+     *    3-1-2 else if start of timetable is equal of start of timetables
+     *      3-1-2-1 throws an exception END IF
      *
-     * 5-if the size of students in timetable is bigger than capacity of timetable
-     *  5-1- throws an exception
-     * 2-read all students of timetable -->allStudents:List<Students>
-     * 3-read a student by getById -->student: Student
-     * 4-add studentId to this List
-     * 6-set student to the timetable
+     * 4-if the size of students in timetable is bigger than capacity of timetable
+     *  4-1- throws an exception
+     * 5-read all students of timetable -->allStudents:List<Students>
+     * 6-read a student by getById -->student: Student
+     * 7-add studentId to this List
+     * 8-set student to the timetable
      * 9-save this timetable
      */
     @Override
     public void addStudentToTimetable(long timetableId, long studentId) {
-        Timetable timetable = getById(timetableId);
+        Timetable target = getById(timetableId);
+        validateStartAndEndOfTimetableWithOtherTimetables(studentId, target);
+        validateTargetTimetableCapacity(target);
+        addStudentToTargetTimetable(studentId, target);
+        timeTableRepository.save(target);
+    }
 
+    public void validateStartAndEndOfTimetableWithOtherTimetables(Long studentId, Timetable target) {
         List<Timetable> timetablesByStudentId = getTimetablesByStudentId(studentId);
 
-        for (Timetable timetables : timetablesByStudentId) {
-            if (timetable.getDate().equals(timetables.getDate())) {
-                if (timetable.getStart().isBefore(timetables.getEnd()) && timetable.getStart().isAfter(timetables.getStart()) || timetable.getEnd()
-                        .isAfter(timetables.getStart()) && timetable.getEnd().isBefore(timetables.getEnd())) {
-                    throw new IllegalArgumentException("This timetable has conflict with other timetables");
-                } else if (timetable.getStart().equals(timetables.getStart())) {
-                    throw new IllegalArgumentException("there is exactly the same timetable");
+        for (Timetable timetable : timetablesByStudentId) {
+            if (target.getDate().equals(timetable.getDate())) {
+
+                boolean startOfTargetBetweenATimetable = target.getStart().isBefore(timetable.getEnd()) && target.getStart().isAfter(timetable.getStart());
+                boolean endOfTargetBetweenAnotherATimetable = target.getEnd().isAfter(timetable.getStart()) && target.getEnd().isBefore(timetable.getEnd());
```

Comment:
Create a method that accepts a LocalDate and a timetable and returns true if the LocalTime is not between start and end of the timetable.
Then both of these lines can call that method.